src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -452,6 +452,10 @@ abstract class ManagedStrategy(
         check(threadId == threadScheduler.scheduledThreadId)
         // check if live-lock is detected
         val decision = loopDetector.visitCodeLocation(threadId, codeLocation)
+        // in case of live-lock, try to abort execution
+        if (decision is LoopDetector.Decision.LivelockThreadSwitch) {
+            tryAbortingUserThreads(threadId, BlockingReason.LiveLocked)
+        }
```

Comment:
Can we move it into `processLoopDetectorDecision` ?

Reply:
Call to `tryAbortingUserThreads` should be done before `onSwitchPoint` call. If adding `tryAbortingUserThreads` inside `processLoopDetectorDecision`, then I would also need to split calls to `onSwitchPoint` -- one in the process loop detector, and the second one before calling `shouldSwitch` after the check for `decision !is LoopDetector.Decision.Idle`